lib/rex/parser/appscan_nokogiri.rb

```diff
@@ -187,31 +187,18 @@ def report_service_info
 
     def record_request_and_response
       return unless(in_issue && has_text)
-      return unless @state[:web_site]
+      return unless @state[:web_site].present?
       really_original_traffic = unindent_and_crlf(@text)
-      split_traffic = really_original_traffic.split(/\r\n\r\n/)
-      request_headers_text = split_traffic.first
-      content_length = 0
-      if request_headers_text =~ /\ncontent-length:\s+([0-9]+)/mni
-        content_length = $1.to_i
-      end
-      if(content_length > 0) and (split_traffic[1].to_s.size >= content_length)
-        request_body_text = split_traffic[1].to_s[0,content_length]
-      else
-        request_body_text = nil
-      end
-      response_headers_text = split_traffic[1].to_s[content_length,split_traffic[1].to_s.size].lstrip
-      request = request_headers_text
-      return unless(request && response_headers_text)
-      response_body_text = split_traffic[2]
+      request_headers, request_body, response_headers, response_body = really_original_traffic.split(/\r\n\r\n/)
+      return unless(request_headers && response_headers)
```

Comment:
request_body and response_body are presumed to be non-nil below, so they should also be guarded here.